Refactor aggregation logic and allow it to be applied to the matrix again

JackWilb opened the pull request:
Closes #238
Closes #232
Closes #225
Relates to #224
Closes #137
Closes #118 (since we've refactored the whole thing)

TODOs:
- [x] Allow matrix to be reset
- [x] Show child legend
- [x] Debug issue when turning off aggregation more than once

gotdairyya commented:
Found one major bug (though seems like it is an easy fix) + a couple of minor ones (that are more opinions than required fixes). Otherwise, everything is working as expected!

**Major bug**
If you submit a connectivity query, view it using the aggregation, then submit another query, enable aggregation is not reset.  I _think_ it should be as simple as adding `store.dispatch.updateEnableAggregation(false)` to line 189 in `ConnectivityQuery.vue`

**Minor Bugs**
(1) If you hover over the + icon, the tooltip shows up very far away
![image](https://user-images.githubusercontent.com/8480871/124813149-c3fd3c00-df21-11eb-9a10-9b1509bf4e21.png)

(2) I would assume if you aggregate by a categorical variable you would see it as an attribute on the parent node. In the example below, the Label attribute shows up as `-` for the aggregated row. But I don't think it's a big deal, it just means you can't filter based on parent node attributes .. 
![image](https://user-images.githubusercontent.com/8480871/124813246-e2fbce00-df21-11eb-8dcc-8eb35d3f1052.png)

(3) Can we update the child class text color to be slate gray so that it is more visually distinct?

JackWilb commented:
Thanks for such thorough testing.

> **Major bug**
> If you submit a connectivity query, view it using the aggregation, then submit another query, enable aggregation is not reset. I _think_ it should be as simple as adding `store.dispatch.updateEnableAggregation(false)` to line 189 in `ConnectivityQuery.vue`

Your suggested fix seemed to work here. 8a5785f

> **Minor Bugs**
> (1) If you hover over the + icon, the tooltip shows up very far away
> ![image](https://user-images.githubusercontent.com/8480871/124813149-c3fd3c00-df21-11eb-9a10-9b1509bf4e21.png)

I think this is a general bug in the rendering of the tooltip. I'll address it in a follow up #289 

> (2) I would assume if you aggregate by a categorical variable you would see it as an attribute on the parent node. In the example below, the Label attribute shows up as `-` for the aggregated row. But I don't think it's a big deal, it just means you can't filter based on parent node attributes ..
> ![image](https://user-images.githubusercontent.com/8480871/124813246-e2fbce00-df21-11eb-8dcc-8eb35d3f1052.png)

Addressed this in 0f21621.

> (3) Can we update the child class text color to be slate gray so that it is more visually distinct?

Fixed in 31fe5e7

> (4) Color scale of child legend doesn't match the child cells in the matrix

I'll fix in #290

> (5) Th z-index of lineup is higher than the tooltips, so sometimes it blocks the tooltip.

I'll add this to #289.

gotdairyya reviewed (approved):
Looks great! Thanks Jack for addressing my concerns!
